<?xml version="1.0" ?>
<launch>

    <env name="GAZEBO_MODEL_PATH" value="${GAZEBO_MODEL_PATH}:$(find generic_gazebo_camera_plugin)/models"/>
    <env name="GAZEBO_RESOURCE_PATH" value="${GAZEBO_RESOURCE_PATH}:$(find generic_gazebo_camera_plugin)/models"/>

    <!-- Gazebo parameters -->
     <arg name="launch_gazebo" default="true"/>
    <arg name="launch_gazebo_gui" default="true"/>
    <arg name="gazebo_world_name" default="$(find generic_gazebo_camera_plugin)/worlds/chessboard.world"/>
    <arg name="launch_rviz" default="true"/>

    <!-- Path to the cameras config file -->
    <arg name="cameras_config_file" default="$(find generic_gazebo_camera_plugin)/config/default.yaml"/> 
    
    <!-- Launch the spawning routine -->
    <node name="spawn_cameras" pkg="generic_gazebo_camera_plugin" type="spawn_cameras.py" output="screen">
        <param name="cameras_config_file" value="$(arg cameras_config_file)" />
    </node>

    <!-- Launch Gazebo -->
    <include if="$(eval launch_gazebo)" file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg gazebo_world_name)"/>
        <arg name="debug" value="false" />
        <arg name="verbose" value="false" />
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="gui" value="$(arg launch_gazebo_gui)"/>     
    </include>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rviz_config_file)" if="$(arg launch_rviz)"/>
    
</launch>
